test/utility/parse_opts.c

```diff
@@ -48,13 +48,88 @@ extern int __wt_optreset;
     } while (0)
 
 /*
- * parse_seed --
- *     Parse a random number generator seed from the command line.
+ * parse_number --
+ *     Parse a number from the command line config string.
  */
 static void
-parse_seed(uint64_t *seedp, const char *seed_str, char **parse_end)
+parse_number(uint64_t *config_key, const char *parse_str, char **parse_end)
 {
-    *seedp = (uint64_t)strtoll(seed_str, parse_end, 10);
+    *config_key = (uint64_t)strtoll(parse_str, parse_end, 10);
+}
+
+/*
+ * parse_tiered_comma_separated_options --
+ *     Parse a command line option for the tiered storage configurations.
+ */
+static void
+parse_tiered_comma_separated_options(const char *config_str, uint64_t *opts_key1,
+  uint64_t *opts_key2, const char *usage, bool insert_delays)
+{
+    char *parse_end;
+    const char *config_end;
+
+    if (*opts_key1 != 0)
+        testutil_die(
+          EINVAL, "%s x,x option specified twice: %s", insert_delays ? "-Pd" : "-Pe", usage);
+
+    /* Point the end of string either to comma or the final null character. */
+    config_end = strchr(config_str, ',');
+
+    if (config_end == NULL) {
+        config_end = &config_str[strlen(config_str)];
+        /*
+         * The delay milliseconds is set to 100 by default for artificial delays and set to 0 for
+         * artificial errors.
+         */
+        *opts_key2 = insert_delays ? 100 : 0;
+    }
+
+    parse_end = (char *)config_end;
+    parse_number(opts_key1, config_str, &parse_end);
+
+    config_str = parse_end;
+    if (*config_str == ',')
+        ++config_str;
+
+    if ((strchr(config_str, ',')) != NULL)
+        testutil_die(EINVAL, "Multiple comma separated values specified, Usage : %s", usage);
+
+    config_end = &config_str[strlen(config_str)];
+    parse_end = (char *)config_end;
+
+    /* Reached end of the string and nothing to parse. */
+    if (*parse_end == '\0' && *config_str == '\0')
```

Comment:
Isn't `*parse_end == '\0' ` always true?  If so, it should be removed from the conditional expression.

Reply:
Thanks @ddanderson. Yes, `*parse_end == '\0'` is not needed and also I fixed a minor bug [here](https://github.com/wiredtiger/wiredtiger/pull/8714/files#diff-e5a8442c87c07a28e1933f4bfa4d89ab1a9ace1b4cd746fa61649e96354cbe5aR82) where we were setting the delay_ms or error_ms to 0 if the command line options are specified and second option is not specified after the comma, lets say -Pd 2, or -Pe 2, .